### PR TITLE
Move more cluster creation code to NewCluster()

### PR DESCRIPTION
**Pull request by johngmyers**

### CODE REVIEW

File: cmd/kops/create_cluster.go [hunk 1]
```diff
@@ -400,17 +388,19 @@ func RunCreateCluster(ctx context.Context, f *util.Factory, out io.Writer, c *Cr
...
+				return err
+			}
 		}
-	}
 
-	if cluster != nil {
-		return fmt.Errorf("cluster %q already exists; use 'kops update cluster' to apply changes", c.ClusterName)
+		if cluster != nil {
+			return fmt.Errorf("cluster %q already exists; use 'kops update cluster' to apply changes", c.ClusterName)
+		}
```
Comment (hakman) [inline cmd/kops/create_cluster.go:403]: Any special reason why this gets its own code block?

Comment (johngmyers) [reply cmd/kops/create_cluster.go:403]: To take `cluster` out of scope from subsequent code.

Comment (hakman) [reply cmd/kops/create_cluster.go:403]: 👍

**Pull request merged**